server.js

```diff
@@ -1,28 +1,133 @@
 import express from 'express'
-import bodyParser from 'body-parser'
+import dotenv from "dotenv";
 import cors from 'cors'
 import mongoose from 'mongoose'
+import listEndpoints from 'express-list-endpoints'
+
+dotenv.config()
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true, useCreateIndex: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
-const port = process.env.PORT || 8080
+const port = process.env.PORT || 9000
 const app = express()
 
+//CREATES SCHEMA
+const thoughtSchema = new mongoose.Schema({
+  message: {
+    type: String,
+    required: [true, "Ding, dong! Message is required!"],
+    validate: {
+      validator: (value) => {
+        return /^[^0-9]+$/.test(value);
+      },
+      message: "Numbers are not allowed",
+    },
+    minlength: [5, 'Hey, the minimum length is 5 characters'],
+    maxlength: [140, 'Hey, the maximum length is 140 characters'],
+    unique: true,
+    trim: true
+    },
+  hearts: {
+    type: Number,
+    default: 0
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now //also () => Date.now()
+  }
+})
+//CREATES SINGLE THOUGHT MODEL
+const Thought = mongoose.model('Thought', thoughtSchema)
+
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
-app.use(bodyParser.json())
+app.use(express.json())
+
+app.get('/', (req, res) => {
+  res.send(listEndpoints(app))
+})
 
-// Start defining your routes here
+// DEFINING ROUTES AND ENDPOINTS
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send("Hola Mundo! This is Estefania's Happy Coding Thoughts API <3")//replaced by listEndpoints
 })
 
+//GET /thoughts: ENDOPINT TO SHOW 20 RECENT POSTS
+app.get("/thoughts", async (req, res) => {
+  const allThoughts = await Thought.find()
+    .sort({ createdAt: -1 })
+    .limit(20)
+    .exec();
+  res.json(allThoughts);
+});
+
+
+//ENDPOINT TO POST A THOUGHT
+app.post("/thoughts", async (req, res) => {
+  try {
+    const newThought = await new Thought(req.body).save();
+    res.json(newThought);
+  } catch (error) {
+    if (error.code === 11000) {
+      res
+        .status(400)
+        .json({ error: "Duplicated value", fields: error.keyValue });
+    }
+    res.status(400).json(error);
+  }
+});
+
+
+//ENDPOINT TO DELETE A THOUGHT
+app.delete('/thoughts/:thoughtId', async (req, res) => {
+  const { thoughtId } = req.params
+
+  try {
+    const deletedThought = await Thought.findByIdAndDelete(id)
+    if (deletedThought) {
+      res.json(deletedThought)
+    } else {
+      res.status(404).json({ message: 'Not found' })
+    }
+
+  } catch (error) {
+    res.status(400).json({ message: 'Invalid request', error })
+  }
+})
```

Comment:
Remember to restrict access to this endpoint because not all users should have access to this functionality 🤓